x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -0,0 +1,368 @@
+        PutPipelineRequest putPipelineRequest = new PutPipelineRequest(ENT_SEARCH_INGESTION_PIPELINE_NAME, bytes, XContentType.JSON);
+        client.admin().cluster().putPipeline(putPipelineRequest);
+        logger.info("Created pipeline" + ENT_SEARCH_INGESTION_PIPELINE_NAME);
+    }
+
+    private static void ensureInternalIndex(Client client, String index, XContentBuilder mappings, Settings settings, String alias) {
+        GetIndexRequest getIndexRequest = new GetIndexRequest().indices(CONNECTOR_CONCRETE_INDEX_NAME);
+        ThreadContext threadContext = client.threadPool().getThreadContext();
+        executeAsyncWithOrigin(threadContext, ENT_SEARCH_ORIGIN, getIndexRequest, new ActionListener<GetIndexResponse>() {
+            public void onResponse(GetIndexResponse getIndexResponse) {
+                logger.info("Found " + index + " index.");
+            }
+
+            public void onFailure(Exception e) {
+                final Throwable cause = ExceptionsHelper.unwrapCause(e);
+                if (cause instanceof ResourceNotFoundException) {
+                    logger.error("Index " + index + " not found.");
+                    createInternalIndex(client, index, mappings, settings, alias);
+                } else {
+                    logger.error("Error getting " + index + " index " + e.toString());
+                }
+            }
+        }, client.admin().indices()::getIndex);
+    }
+
+    private static void ensurePipeline(Client client) {
+        GetPipelineRequest getPipelineRequest = new GetPipelineRequest(ENT_SEARCH_INGESTION_PIPELINE_NAME);
+        ThreadContext threadContext = client.threadPool().getThreadContext();
+        executeAsyncWithOrigin(threadContext, ENT_SEARCH_ORIGIN, getPipelineRequest, new ActionListener<>() {
+            public void onResponse(GetPipelineResponse getPipelineResponse) {
+                if (getPipelineResponse.isFound() == false) {
+                    createPipeline(client);
+                } else {
+                    logger.info("Found " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " pipeline.");
+                }
+            }
+
+            public void onFailure(Exception e) {
+                final Throwable cause = ExceptionsHelper.unwrapCause(e);
+                if (cause instanceof ResourceNotFoundException) {
+                    logger.error("Pipeline " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " not found.");
+                    createPipeline(client);
+                } else {
+                    logger.error("Error getting " + ENT_SEARCH_INGESTION_PIPELINE_NAME + " index " + e.toString());
+                }
+            }
+        }, client.admin().cluster()::getPipeline);
+    }
+
+    private static Settings getConnectorsIndexSettings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-3")
```

Comment:
CC @ioanatia. I'm reasonably sure that these settings were picked pretty explicitly a while ago, and I think Ioana helped with that.

Reply:
I was wrong on the `SETTING_NUMBER_OF_REPLICAS` set to 0, it's needed to ensure that the cluster stay green if we have a single node.
Regarding the auto expand replica, please note that `0-1` is what we use for all system indices so unless we have a good reason to deviate we should do the same.

Reply:
Found the old context:
1. we picked `0-3` originally because [that's what we'd been using for our Actastic indices](https://github.com/elastic/enterprise-search-team/issues/2199#issue-1287548468)
2. @ioanatia  [confirmed](https://github.com/elastic/enterprise-search-team/issues/2199#issuecomment-1214705139) that this was a good setting for these new indices.